Sources/Datadog/Tracing/TracingFeature.swift

```diff
@@ -14,13 +19,156 @@ internal final class TracingFeature {
 
     // MARK: - Dependencies
 
+    let appContext: AppContext
     let dateProvider: DateProvider
+    let tracingUUIDGenerator: TracingUUIDGenerator
+    let userInfoProvider: UserInfoProvider
+    let networkConnectionInfoProvider: NetworkConnectionInfoProviderType
+    let carrierInfoProvider: CarrierInfoProviderType
+
+    // MARK: - Components
+
+    /// Span files storage.
+    let storage: Storage
+    /// Spans upload worker.
+    let upload: Upload
+
+    /// Encapsulates  storage stack setup for `TracingFeature`.
+    class Storage {
+        /// Writes spans to files.
+        let writer: FileWriter
+        /// Reads spans from files.
+        let reader: FileReader
+
+        init(
+            directory: Directory,
+            performance: PerformancePreset,
+            dateProvider: DateProvider,
+            readWriteQueue: DispatchQueue
+        ) {
+            let orchestrator = FilesOrchestrator(
+                directory: directory,
+                writeConditions: WritableFileConditions(performance: performance),
+                readConditions: ReadableFileConditions(performance: performance),
+                dateProvider: dateProvider
+            )
+
+            // NOTE: any change to tracing data format requires updating the tracing directory url to be unique
+            let dataFormat = DataFormat(
+                prefix: "",
+                suffix: "",
+                separator: "\n"
+            )
```

Comment:
maybe we can pass `DataFormat` to `obtainTracingFeatureDirectory` function in order not to rely on a in-code comment only
```swift
func obtainTracingFeatureDirectory(for dataFormat: DataFormat) -> Directory {
  let hash = dataFormat.deterministicHash
  let directory = {a path with hash value}
  return directory
}
```
`Hashable` doesn't work for this case as it randomizes seeds at launch.
i'm leaving this idea here only, we may improve in the future

Reply:
Yes, using hash function seems interesting, but it's not much practical when it comes to data cleanup when format changes (when we change it, the SDK should do `previousFormatDirectories.forEach { directory in directory.delete() }`.

I was rather thinking `let version: Int` on the `DataFormat` itself. This would be more correct, as it also enables migration to any past format in the next version (`A` → `B` → `A`).

Anyway, I decided to keep it simple, as we don't yet need any migration nor format change. Also, I'm not sure how the upcoming RUM format will look like, so IMO simple is better for now.